themes/finna2/templates/myresearch/menu.phtml

```diff
@@ -5,229 +5,213 @@
-      <a href="<?=$this->url('myresearch-fines')?>" class="list-group-item" aria-current="<?= $this->active == 'fines' ? 'page' : 'false' ?>">
-        <?=$this->transEsc('Fines')?>
-      </a>
-    </li>
-    <?php if ($user && $user->libraryCardsEnabled()): ?>
-      <li class="menu-library-cards hidden-xs hidden-sm<?=$this->active == 'librarycards' ? ' active' : ''?>" >
-        <a href="<?=$this->url('librarycards-home')?>" class="list-group-item" aria-current="<?= $this->active == 'librarycards' ? 'page' : 'false' ?>">
-          <?=$this->transEsc('Library Cards')?>
-        </a>
-      </li>
-    <?php endif; ?>
-  <?php endif; ?>
-<?php endif; ?>
-<?php if ('ils-none' !== $this->ils()->getOfflineMode()): ?>
-  <?php if ($this->ils()->checkCapability('getMyProfile')): ?>
-    <li class="menu-profile hidden-xs hidden-sm<?=$this->active == 'profile' ? ' active' : ''?>">
-      <a href="<?=$this->url('myresearch-profile')?>" class="list-group-item" aria-current="<?= $this->active == 'profile' ? 'page' : 'false' ?>">
-        <?=$this->transEsc('Profile')?>
-      </a>
-    </li>
-  <?php endif; ?>
-<?php endif; ?>
-<li class="menu-history hidden-xs hidden-sm <?=$this->active == 'history' ? ' active' : ''?>">
-  <a href="<?=$this->url('search-history')?>?require_login" class="list-group-item" aria-current="<?= $this->active == 'history' ? 'page' : 'false' ?>">
-    <?=$this->transEsc('history_saved_searches')?>
-  </a>
-</li>
 
-<?php if ($this->userlist()->getMode() !== 'disabled'): ?>
-  <li class="menu-favorites mylist-menu-favorites <?= $listOpen ? ' active' : ''?>">
-    <a href="<?=$this->url('myresearch-favorites')?>" id="open-list" class="<?= !$listOpen ? 'collapsed ' : '' ?>list-group-item hidden-xs hidden-sm" aria-current="<?= $this->active == 'favorites' ? 'page' : 'false' ?>">
-      <?=$this->transEsc('Favorites')?>
-      <i class="hidden fa fa-spinner fa-spin"></i>
-      <span class="ajax-error hidden">
-        <small> <?=$this->translate('An error has occurred'); ?></small>
-      </span>
-      <span class="caret"></span>
-    </a>
-    <div class="mylist-bar">
-      <?php if($listOpen) : ?>
-        <?= $this->partial('myresearch/mylist-navi.phtml', ['user' => $user, 'isDefaultList' => $isDefaultList, 'activeId' => $activeId, 'activePublic' => $activePublic, 'lists' => $lists, 'results' => $this->results]);  ?>
-      <?php endif; ?>
-    </div>
-  </li>
-<?php endif; ?>
-<li class="menu-logout hidden-xs hidden-sm">
-  <a href="<?=$this->url('myresearch-logout')?>" class="list-group-item">
-    <?=$this->transEsc('Log Out')?>
-    <i class="pull-right fa fa-sign-out fa-lg" aria-hidden="true"></i>
-  </a>
-</li>
-<?php $menuEntries = ob_get_clean(); ?>
+    $menuListItems = [];
+    if ($showLibraryTabs && !$ilsOffline) {
+      $menuListItems = [
+        'checkedout' => [
+          'url' => $this->url('myresearch-checkedout'),
```

Comment:
Jätä pois $this->url -kutsut taulukon muodostamisesta. Laita vaikka avaimeksi 'route' ja kutsu sitten $this->url() routelle vasta kun linkkiä rendataan.